### ISSUE TITLE
mutate on createResource does not update state

### ISSUE THREAD

**Issue by Kresten** [solidjs/solid#1864]
### Describe the bug

`mutate` returned from `createResource` is `setValue` ([line 723 here](https://github.com/solidjs/solid/blob/f79ba4d9087d74ef7e55fd0dc336c98ce2ec8526/packages/solid/src/reactive/signal.ts#L723)) and is oblivious to the loading-state of the resource.

This is an issue, when we use custom state handlers reacting on state while mutating signals in resources.

### Your Example Website or App

https://stackblitz.com/edit/solidjs-templates-uwtyzi?file=src%2FApp.tsx

### Steps to Reproduce the Bug or Issue

```
  const [resource, { mutate }] = createResource(
    () => undefined,
    async () => {
      return 'test'
    }
  )
  mutate('updated')
  resource.latest // 'updated'
  resource.state // 'unresolved'

  const [resource2, { mutate: mutate2 }] = createResource(async () => {
    throw new Error('Boom')
  })
  setTimeout(() => mutate2('updated'), 1000)
  console.log(resource2.latest) // 'updated'
  console.log(resource2.state) // 'errored'
```

### Expected behavior

As a user I expect a way to set the state after mutation, e.g. to `ready` after I have mutated the data.

### Platform

- OS: [ macOS]
- Browser: [ Chrome]
- Version: [116]

**Comment by shawnyu5** [comment]
Running into a similar issue here. When I use `mutate` to manually update a resource, the update does not trigger a re-render.

**Comment by elite174** [comment]
The same thing is when you trigger a resource inside data function when searchParams are being changed.

![image](https://github.com/solidjs/solid/assets/13636224/efdda460-5b0c-4b56-8ac3-78898c446011)

When you do navigation `createEffect(() => console.log(comments.state));` is always `ready`
![image](https://github.com/solidjs/solid/assets/13636224/27756102-d1f1-46ab-979c-de8b0e4d9214)

**Comment by elite174** [comment]
Oh, interesting thing
when I do `createComputed(() => console.log(comments.state));` I see 

`refresing`
`ready`

I suppose this is because of transition happening in the router. However I need to reflect this state in the UI, but UI works like an effect....

**Comment by elite174** [comment]
How can I reflect state of resource in the UI in transition?

**Comment by ryansolid** [comment]
I have a fix for this. The one that always should have been. I do wonder if people were using this in unexpected ways though. The solution is have the mutate call actually call load complete. But it means the setter is no longer transparently sent back. For normal operation this is fine, but I wonder if people might have tried to leverage this with custom signal stores.

------------------

> How can I reflect state of resource in the UI in transition?

There was another issue about this. Generally you can't because the rest of the app doesn't see it. Your best bet is to use the `isRouting` or `isPending` state from the transition to know if it is in flight. I sort of regret adding all these states into the resources because the other mechanisms manage it.

**Comment by elite174** [comment]
Thank you for the answer! Will come up with some ideas about potential workarounds. I did use `isRouting` before, however I faced a "bug" when you navigate to another page, the styles for "loading" state are applied, however here's completely different semantics, because we need to apply them only when the resource is being refreshed. That's why I decided to go with resource state, but unfortunately it doesn't work as I'd expect.

**Comment by elite174** [comment]
Reproduction example: https://playground.solidjs.com/anonymous/181aed31-efe8-492b-a8b7-e15f43e9a123

**Comment by ryansolid** [comment]
Yeah we won't be changing createResources semantics this late in 1.x. So closing as not planned/stale.
